classes/webservice/WebserviceOutputBuilder.php

```diff
@@ -378,131 +383,92 @@ public function getContent($objects, $schema_to_display = null, $fields_to_displ
         }
 
-        return $output;
+        return $this->objectRender->renderNode($this->output);
     }
 
     /**
      * Create the tree diagram with no details.
      *
      * @param ObjectModel $object create by the entity
-     * @param int $depth the depth for the tree diagram
-     *
-     * @return string
      */
-    public function renderEntityMinimum($object, $depth)
+    public function renderEntityMinimum($object)
     {
-        $class = get_class($object);
-        if (!isset(WebserviceOutputBuilder::$_cache_ws_parameters[$class])) {
-            WebserviceOutputBuilder::$_cache_ws_parameters[$class] = $object->getWebserviceParameters();
-        }
-        $ws_params = WebserviceOutputBuilder::$_cache_ws_parameters[$class];
+        $ws_params = $this->loadWsParams($object);
 
         $more_attr['id'] = $object->id;
-        $more_attr['xlink_resource'] = $this->wsUrl . $ws_params['objectsNodeName'] . '/' . $object->id;
-        $output = $this->setIndent($depth) . $this->objectRender->renderNodeHeader($ws_params['objectNodeName'], $ws_params, $more_attr, false);
+        $more_attr['xlink:href'] = self::$wsUrl . $ws_params['objectsNodeName'] . '/' . $object->id;
 
-        return $output;
+        $this->output->addParentNode($ws_params['objectNodeName'], $more_attr);
     }
 
     /**
      * Build a schema blank or synopsis.
      *
      * @param ObjectModel $object create by the entity
      * @param array $ws_params webserviceParams from the entity
-     *
-     * @return string
      */
     protected function renderSchema($object, $ws_params)
     {
-        $output = $this->objectRender->renderNodeHeader($ws_params['objectNodeName'], $ws_params);
+        $parentNode = $this->output->addParentNode($ws_params['objectNodeName'], $ws_params);
+
         foreach ($ws_params['fields'] as $field_name => $field) {
-            $output .= $this->renderField($object, $ws_params, $field_name, $field, 0);
+            $this->renderField($parentNode, $object, $ws_params, $field_name, $field);
         }
         if (isset($ws_params['associations']) && count($ws_params['associations']) > 0) {
             $this->fieldsToDisplay = 'full';
-            $output .= $this->renderAssociations($object, 0, $ws_params['associations'], $ws_params);
+            $this->renderAssociations($parentNode, $object, $ws_params['associations'], $ws_params);
         }
-        $output .= $this->objectRender->renderNodeFooter($ws_params['objectNodeName'], $ws_params);
-
-        return $output;
     }
 
     /**
      * Build the entity detail.
      *
+     * @param ApiNode $parentNode node to which append entity
      * @param ObjectModel $object create by the entity
-     * @param int $depth the depth for the tree diagram
      *
-     * @return string
+     * @return void
      */
-    public function renderEntity($object, $depth)
+    public function renderEntity(ApiNode &$parentNode, ObjectModel $object): void
```

Comment:
Could you document the BC Break ?

Reply:
Hello, changing the signature of a public method is considered a BC break, any third party element relying on this method would be broken

so even if its' unlikely, we document the BC break ;)

Reply:
All BC breaks documented in description as advised. The list is pretty huge :)